static/app/views/dashboardsV2/datasetConfig/context.tsx

```diff
@@ -0,0 +1,127 @@
+      name: 'DatasetConfigContext',
+    });
+
+  function DatasetConfigProvider({children}: {children: ReactNode}) {
+    const [datasetConfig, setDatasetConfig] = useState<
+      DatasetConfig<SeriesResponse, TableResponse> | undefined
+    >(undefined); // undefined means not initialized
+
+    return (
+      <_DatasetConfigProvider
+        value={{
+          datasetConfig,
+          setDatasetConfig,
+        }}
+      >
+        {children}
+      </_DatasetConfigProvider>
+    );
+  }
+  const DatasetConfigConsumer = DatasetConfigContext.Consumer;
+
+  return {
+    DatasetConfigProvider,
+    useDatasetConfigContext,
+    DatasetConfigContext,
+    DatasetConfigConsumer,
+  };
+}
+
+export const {
+  DatasetConfigProvider: ReleasesDatasetConfigProvider,
+  useDatasetConfigContext: useReleasesDatasetConfigContext,
+  DatasetConfigContext: ReleasesDatasetConfigContext,
+  DatasetConfigConsumer: ReleasesDatasetConfigConsumer,
+} = createDatasetConfigContext<
+  SessionApiResponse | MetricsApiResponse,
+  SessionApiResponse | MetricsApiResponse
+>();
+
+export const {
+  DatasetConfigProvider: IssuesDatasetConfigProvider,
+  useDatasetConfigContext: useIssuesDatasetConfigContext,
+  DatasetConfigContext: IssuesDatasetConfigContext,
+  DatasetConfigConsumer: IssuesDatasetConfigConsumer,
+} = createDatasetConfigContext<never, Group[]>();
+
+export const {
+  DatasetConfigProvider: ErrorsAndTransactionsDatasetConfigProvider,
+  useDatasetConfigContext: useErrorsAndTransactionsDatasetConfigContext,
+  DatasetConfigContext: ErrorsAndTransactionsDatasetConfigContext,
+  DatasetConfigConsumer: ErrorsAndTransactionsDatasetConfigConsumer,
+} = createDatasetConfigContext<
+  EventsStats | MultiSeriesEventsStats,
+  TableData | EventsTableData
+>();
+
+export function getDatasetConfigProvider(widgetType: WidgetType | undefined) {
+  switch (widgetType) {
+    case WidgetType.RELEASE:
+      return ReleasesDatasetConfigProvider;
+    case WidgetType.ISSUE:
+      return IssuesDatasetConfigProvider;
+    case WidgetType.DISCOVER:
+    default:
+      return ErrorsAndTransactionsDatasetConfigProvider;
+  }
+}
+
+export function getDatasetConfigContext(widgetType: WidgetType | undefined) {
+  switch (widgetType) {
+    case WidgetType.RELEASE:
+      return ReleasesDatasetConfigContext;
+    case WidgetType.ISSUE:
+      return IssuesDatasetConfigContext;
+    case WidgetType.DISCOVER:
+    default:
+      return ErrorsAndTransactionsDatasetConfigContext;
+  }
+}
+
+export function getDatasetConfigConsumer(widgetType: WidgetType | undefined) {
+  switch (widgetType) {
+    case WidgetType.RELEASE:
+      return ReleasesDatasetConfigConsumer;
+    case WidgetType.ISSUE:
+      return IssuesDatasetConfigConsumer;
+    case WidgetType.DISCOVER:
+    default:
+      return ErrorsAndTransactionsDatasetConfigConsumer;
+  }
+}
+
+export function getDefinedContext(widgetType: WidgetType | undefined) {
```

Comment:
should we name this something like `getDefinedContextHook` or `getDefinedUseContextHook` or something similar? I feel like this function could confused with `getDatasetConfigContext` since they sound very similar